Modules/Relationships/test/Relationships.Application.Tests/Tests/Relationships/Commands/TerminateRelationship/HandlerTests.cs

```diff
@@ -0,0 +1,121 @@
+using Backbone.UnitTestTools.Data;
+using FakeItEasy;
+using FluentAssertions;
+using Xunit;
+
+namespace Backbone.Modules.Relationships.Application.Tests.Tests.Relationships.Commands.TerminateRelationship;
+
+public class HandlerTests
+{
+    [Fact]
+    public async Task Returns_the_updated_relationship()
+    {
+        // Arrange
+        var activeIdentity = TestDataGenerator.CreateRandomIdentityAddress();
+        var activeDevice = TestDataGenerator.CreateRandomDeviceId();
+
+        var fakeRelationshipsRepository = A.Fake<IRelationshipsRepository>();
+        var relationship = TestData.CreateActiveRelationship(from: activeIdentity);
+        A.CallTo(() => fakeRelationshipsRepository
+            .FindRelationship(relationship.Id, activeIdentity, A<CancellationToken>._, true)).Returns(relationship);
+
+        var fakeUserContext = A.Fake<IUserContext>();
+        A.CallTo(() => fakeUserContext.GetAddress()).Returns(activeIdentity);
+        A.CallTo(() => fakeUserContext.GetDeviceId()).Returns(activeDevice);
+
+        var handler = CreateHandler(fakeUserContext, fakeRelationshipsRepository);
+
+        // Act
+        var response = await handler.Handle(new TerminateRelationshipCommand
+        {
+            RelationshipId = relationship.Id
+        }, CancellationToken.None);
+
+        // Assert
+        response.Id.Should().NotBeNull();
+        response.Status.Should().Be(RelationshipStatus.Terminated);
+        response.AuditLog.Should().HaveCount(3);
+    }
+
+    [Fact]
+    public async Task Saves_the_updated_relationship()
+    {
+        // Arrange
+        var activeIdentity = TestDataGenerator.CreateRandomIdentityAddress();
+        var activeDevice = TestDataGenerator.CreateRandomDeviceId();
+
+        var mockRelationshipsRepository = A.Fake<IRelationshipsRepository>();
+        var relationship = TestData.CreateActiveRelationship(from: activeIdentity);
+        A.CallTo(() => mockRelationshipsRepository
+            .FindRelationship(relationship.Id, activeIdentity, A<CancellationToken>._, true)).Returns(relationship);
+
+        var fakeUserContext = A.Fake<IUserContext>();
+        A.CallTo(() => fakeUserContext.GetAddress()).Returns(activeIdentity);
+        A.CallTo(() => fakeUserContext.GetDeviceId()).Returns(activeDevice);
+
+        var handler = CreateHandler(fakeUserContext, mockRelationshipsRepository);
+
+        // Act
+        await handler.Handle(new TerminateRelationshipCommand
+        {
+            RelationshipId = relationship.Id
+        }, CancellationToken.None);
+
+        // Assert
+        A.CallTo(
+                () => mockRelationshipsRepository.Update(
+                    A<Relationship>.That.Matches(r => r.Id == relationship.Id && r.Status == RelationshipStatus.Terminated))
+            )
+            .MustHaveHappenedOnceExactly();
+    }
+
+    [Fact]
+    public async Task Publishes_RelationshipStatusChangedIntegrationEvent()
```

Comment:
IF the `RelationshipTerminatedIntegrationEvent` is actually required, you should test that it is published.

Reply:
Changes made.